### ISSUE TITLE
Issues with async tests

### ISSUE THREAD

**Issue by panrafal** [roadhump/karma-mocha-own-reporter#1]
When running nested asynchronous tests (ones returning promises) the reporter seems to end the suite prematurely and even twice - leading to `invalid array length` error.

Below is a part of output for our test suite. `setState` should be nested under `DataProvider`. `handles async data transform`, `handles async chaining` and `emits on state change` all return a promise.

```
    DataProvider
      ✓ handles data transform
    ✓ handles async data transform (43ms)
  ✓ handles chaining after transform (3ms)
  ✓ handles async chaining (16ms)
setState
  ✓ merges objects (7ms)
  ✓ emits on state updates
Missing error handler on `socket`.
RangeError: Invalid array length
```

p.s. Great idea!

**Comment by roadhump** [comment]
It looks like a bug with nested suites, not just async ones. I've fixed it in version 1.1.1, can you check it?

**Comment by panrafal** [comment]
Yup, it works! Thanks!
